lib/dialects/abstract/query-generator.js

```diff
@@ -1493,124 +1182,273 @@ const QueryGenerator = {
       }
     }
 
-    query += ';';
+    return `${query};`;
 
-    return query;
   },
 
-  getQueryOrders(options, model, subQuery) {
-    const mainQueryOrder = [];
-    const subQueryOrder = [];
+  escapeAttributes(attributes, options, mainTableAs) {
 
-    const validateOrder = order => {
-      if (order instanceof Utils.Literal) return;
+    return attributes && attributes.map(attr => {
+      let addTable = true;
 
-      if (!_.includes([
-        'ASC',
-        'DESC',
-        'ASC NULLS LAST',
-        'DESC NULLS LAST',
-        'ASC NULLS FIRST',
-        'DESC NULLS FIRST',
-        'NULLS FIRST',
-        'NULLS LAST'
-      ], order.toUpperCase())) {
-        throw new Error(util.format('Order must be \'ASC\' or \'DESC\', \'%s\' given', order));
+      if (attr._isSequelizeMethod) {
+        return this.handleSequelizeMethod(attr);
       }
+
+      if (Array.isArray(attr) && attr.length === 2) {
+        attr = attr.slice();
+
+        if (attr[0]._isSequelizeMethod) {
+          attr[0] = this.handleSequelizeMethod(attr[0]);
+          addTable = false;
+        } else if (attr[0].indexOf('(') === -1 && attr[0].indexOf(')') === -1) {
+          attr[0] = this.quoteIdentifier(attr[0]);
+        }
+        attr = [attr[0], this.quoteIdentifier(attr[1])].join(' AS ');
+      } else {
+        attr = attr.indexOf(Utils.TICK_CHAR) < 0 && attr.indexOf('"') < 0 ? this.quoteIdentifiers(attr) : attr;
+      }
+
+      if (options.include && attr.indexOf('.') === -1 && addTable) {
+        attr = mainTableAs + '.' + attr;
+      }
+
+      return attr;
+
+    });
+
+  },
+
+  generateInclude(include, parentTableName, topLevelOptions, topLevelTableInfo, topLevelTableAttributes, topLevelSubQuery) {
```

Comment:
perhaps all `topLevel*` could be grouped into a `topLevel` object, if it makes sense.